anonlinkclient/utils.py

```diff
@@ -57,16 +57,29 @@ def generate_candidate_blocks_from_csv(input_f: TextIO,
         msg = 'The schema is not a valid JSON file'
         raise_from(ValueError(msg), e)
 
+    blocking_method = blocking_config['type']
+
     pii_data = []  # type: List[Any]
     # read from clks
-    if blocking_config['config'].get('input-clks', False):
-        pii_data = json.load(input_f)['clks']
-    else:  # read from CSV file
-        reader = unicode_reader(input_f)
-        if header:
-            next(reader)
-        for line in reader:
-            pii_data.append(tuple(element.strip() for element in line))
+    if blocking_method == 'lambda-fold' and blocking_config['config']['input-clks']:
+        try:
+            pii_data = json.load(input_f)['clks']
+        except ValueError:  # since JSONDecodeError is inherited from ValueError
+            raise TypeError('Upload should be CLKs not CSVs')
+
+    # read from CSV file
+    else:
+        # sentinel check for input
+        try:
+            json.load(input_f)
+            raise TypeError('Upload should be CSVs not CLKs')
+        except ValueError:
```

Comment:
what if I provide an input which is not json and not a csv? Should we rather check the file ending instead?

Reply:
I think the likelihood for this to happen is quite low given the context of usage. Another problem is that the input to this function is an `io.TextIO` object with no access to its filename

Reply:
I made a commit to fix this